view/src/components/farms.js

```diff
@@ -358,6 +426,344 @@ class Farms extends Component {
+              value={nameQuery}
+              onSelect={(e) => this.handleStringSearch("nameQuery", e)}
+              fullWidth={true}
+              renderInput={(params) => (
+                <TextField
+                  {...params}
+                  label="Farm Names"
+                  placeholder="Type or Select a Farm Name"
+                  variant="outlined"
+                  name="nameQuery"
+                  onChange={(e) => this.handleStringSearch("nameQuery", e)}
+                  InputProps={{
+                    ...params.InputProps,
+                    startAdornment: (
+                      <>
+                        <InputAdornment position="start">
+                          <SearchIcon />
+                        </InputAdornment>
+                        {params.InputProps.startAdornment}
+                      </>
+                    ),
+                  }}
+                />
+              )}
+            />
+          </AccordionSummary>
+          <AccordionDetails>{this.extraFiltersAccordion()}</AccordionDetails>
+          <Button
+            className={classes.clearSearchButton}
+            onClick={this.resetCards}
+            variant="contained"
+            color="primary"
+            size="medium"
+            startIcon={<ClearIcon />}
+          >
+            Clear
+          </Button>
+        </Accordion>
+      </div>
+    );
+  };
+
+  /**
+   * Returns tag filtering menu for searching cards.
+   * Re-using the autocomplete for the add new/edit farm form
+   * causes issues with setting a conditional value for
+   * Autocomplete value={}, so separating the two is smoother.
+   */
+  tagsAutocomplete = () => {
+    const { classes } = this.props;
+    const { allFarmTags, tagsQuery } = this.state;
+
+    return (
+      <Autocomplete
+        className={classes.searchBars}
+        multiple
+        id="tags-search"
+        onChange={this.handleTagFilter}
+        options={allFarmTags}
+        value={tagsQuery}
+        fullWidth
+        renderTags={(value, getTagProps) =>
+          value.map((option, index) => (
+            <Chip label={option} {...getTagProps({ index })} />
+          ))
+        }
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="outlined"
+            label="Farm Tags"
+            placeholder="Type or Select a Farm Tag"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <>
+                  <InputAdornment position="start">
+                    <SearchIcon />
+                  </InputAdornment>
+                  {params.InputProps.startAdornment}
+                </>
+              ),
+            }}
+          />
+        )}
+      />
+    );
+  };
+
+  /** Updates filteredData with a new array; used with filters.js queries */
+  updateCards = (newValues) => {
+    if (newValues === []) {
```

Comment:
you probably want `if (newValues.length === 0)` or similar, since this is checking if the array is the exact same memory reference as []

(don't forget to fix the other one too)